esr/fitting/match.py

```diff
@@ -147,17 +152,70 @@ def f1(x):
 
             except:
                 negloglike_all[i] = np.nan
-
-            k -= np.sum(Nsteps<1)       # For the codelen, we effectively don't have the parameter that had Nsteps<1
                 
+            if np.isfinite(negloglike_all[i]):
+                k -= np.sum(Nsteps<1)
+                kept_mask = Nsteps>=1
+            else:
+                # Let's see if setting any of the parameters to zero is ok
+                try_idx = np.arange(nparams)[Nsteps < 1]
+                for r in reversed(range(1, len(try_idx))):
+                    for idx in itertools.combinations(try_idx, r):
+                        p = np.copy(ptrue)
+                        p[idx] = 0.
+                        if k==1:
+                            eq_numpy = sympy.lambdify([x, a0], eq, modules=["numpy"])
```

Comment:
This is unnecessary as we are still on the same function, so we don't need to re-sympify the equation as eq_numpy is already the one we're after. Please delete this line.